### ISSUE TITLE
Don't pin the leaf certificate in Postfix configuration

### ISSUE THREAD

**Issue by garrettr** [freedomofpress/securedrop#1076]
The Postfix configuration on the Monitor Server currently uses certificate fingerprint verification for TLS connections to the SMTP relay. As stated in the [docs](http://www.postfix.org/TLS_README.html#client_tls_fprint):

> Fingerprint verification may be feasible for an SMTP "VPN" connecting a small number of branch offices over the Internet, or for secure connections to a central mail hub. It works poorly if the remote SMTP server is managed by a third party, and its public certificate changes periodically without prior coordination with the verifying site.

While we encourage SecureDrop installations to use their own organization's SMTP servers, it is common for them to use public SMTP servers (such as `smtp.gmail.com`) if they cannot use their own. In this case, the instances fall into the "works poorly" camp.

For example, Google rotates the certificate for `smtp.gmail.com` regularly. They appear to do it approximately every 3-4 months, since they just rotated it this morning and our last issue referring to the same problem was from around March 23rd - https://github.com/freedomofpress/securedrop/issues/957#issuecomment-87006791. When they rotate, the certificate's fingerprint changes and subsequent validations fail, which causes sending to be deferred.  This causes all OSSEC alerts to stop being delivered without any warning to the admins, and requires manual intervention from the admins (update the fingerprint in `prod-specific.yml` and re-run the playbook) to resolve.

**Comment by garrettr** [comment]
Aaand Google appears to have switched back to the old cert again, here's a recent re-check:

```sh
~/code/freedomofpress/securedrop on develop 15:52:48
$ openssl s_client -connect smtp.gmail.com:587 -starttls smtp < /dev/null 2>/dev/null | openssl x509 -fingerprint -noout -in /dev/stdin | cut -d'=' -f2
D3:7C:82:FC:D0:5F:8F:D7:DA:A2:59:8C:42:D7:B2:9F:C1:9F:7E:60
```

**Comment by conorsch** [comment]
Depending on location, we've seen all of the following fingerprints within the same 10-minute window:

* `6D:87:EE:CB:D0:37:2F:88:B8:29:06:FB:35:F4:65:00:7F:FD:84:29`
* `D3:7C:82:FC:D0:5F:8F:D7:DA:A2:59:8C:42:D7:B2:9F:C1:9F:7E:60`
* `4F:6D:5C:58:94:28:C1:A4:36:B9:C4:B5:50:09:FC:5F:FB:47:91:BD`

So relying on fingerprint verification of the SMTP relay is no longer a practical solution for sending the OSSEC alerts. We'll need to decide on a better implementation and update the postfix config. It's likely that mismatched fingerprints are preventing alerts from being sent already, so we should provide a solution soon.

**Comment by conorsch** [comment]
After talking with @garrettr and @dolanjs, we want to provide sane default behavior out of the box for admins. If an admin chooses to use Google's SMTP servers, we should trust the cert lookup, rather than expecting a specific fingerprint. Postfix does support reading a [list of fingerprints](http://www.postfix.org/postconf.5.html#smtp_tls_fingerprint_cert_match), rather than just a single one, but that's just kicking the can down the road.

Switching `smtp_tls_security_level` from `fingerprint` to `secure` will require STARTTLS on all connections, and trust the cert lookup, as long as it's validated by the CAs specified in `smtp_tls_CApath`. By default the system CAs should be used.

If necessary, we can add in a variable that will permit an admin to override `smtp_tls_CApath` with a specific `smtp_tls_CAfile`, which should provide adequate opt-in support for folks who want to use a custom cert.

**Comment by dolanjs** [comment]
We should be clear about the difference between `verify` and `secure`. May need to also provide an override for that option depending on the instance smtp relay.

The `verify` option validates the cert against the hostname returned by the DNS lookup for the MX record.

>  any DNS names in the SubjectAlternativeName certificate extension are used to verify the remote SMTP server name. If no DNS names are specified, the certificate CommonName is checked.

> ...

> The peer hostname is verified, but verification is vulnerable to DNS response forgery.

The `secure` options does:

> any DNS names in the SubjectAlternativeName certificate extension are used to verify the remote SMTP server name. If no DNS names are specified, the certificate CommonName is checked

> ...

> The Postfix SMTP client will encrypt all traffic and verify the destination name immune from forged DNS responses. MX lookups are still used to find the hostnames of the SMTP servers for example.com, but these hostnames are not used when checking the names in the server certificate(s). Rather, the requirement is that the MX hosts for example.com have trusted certificates with a subject name of example.com or a sub-domain, see the documentation for the smtp_tls_secure_cert_match parameter.  [1]

[1] http://www.postfix.org/TLS_README.html#client_tls_secure

**Comment by garrettr** [comment]
Google rotated the certificate again; unfortunately, we're not exactly sure when this happened.

```
$ openssl s_client -connect smtp.gmail.com:587 -starttls smtp < /dev/null 2>/dev/null |
    openssl x509 -fingerprint -noout -in /dev/stdin | cut -d'=' -f2
3B:05:7C:2C:22:79:BB:EA:AA:F6:93:DD:21:1C:55:6B:8D:7A:0D:05
```

**Comment by garrettr** [comment]
Closed by #1100.

**Comment by garrettr** [comment]
And, they're back to the old cert:

```
~/code/freedomofpress/securedrop on ⭠ master! ⌚ 11:18:39
$ openssl s_client -connect smtp.gmail.com:587 -starttls smtp < /dev/null 2>/dev/null |
    openssl x509 -fingerprint -noout -in /dev/stdin | cut -d'=' -f2
D3:7C:82:FC:D0:5F:8F:D7:DA:A2:59:8C:42:D7:B2:9F:C1:9F:7E:60
```
